lib/modules/platform/codecommit/codecommit-client.ts

```diff
@@ -0,0 +1,344 @@
+}
+
+export async function getFile(
+  repositoryName: string | undefined,
+  filePath: string,
+  commitSpecifier: string | undefined
+): Promise<GetFileOutput> {
+  const input: GetFileInput = {
+    repositoryName,
+    filePath,
+    commitSpecifier,
+  };
+  const cmd: GetFileCommand = new GetFileCommand(input);
+  return await codeCommitClient.send(cmd);
+}
+
+export async function listPullRequests(
+  repositoryName: string,
+  authorArn: string
+): Promise<ListPullRequestsOutput> {
+  const input: ListPullRequestsInput = {
+    repositoryName,
+    authorArn,
+    pullRequestStatus: PullRequestStatusEnum.OPEN,
+  };
+  const cmd = new ListPullRequestsCommand(input);
+  return await codeCommitClient.send(cmd);
+}
+
+export async function getRepositoryInfo(
+  repository: string
+): Promise<GetRepositoryOutput> {
+  const input: GetRepositoryInput = {
+    repositoryName: `${repository}`,
+  };
+  const cmd = new GetRepositoryCommand(input);
+  return await codeCommitClient.send(cmd);
+}
+
+export async function getPr(
+  pullRequestId: string
+): Promise<GetPullRequestOutput | undefined> {
+  const input: GetPullRequestInput = {
+    pullRequestId,
+  };
+  const cmd = new GetPullRequestCommand(input);
+  let res;
+  try {
+    res = await codeCommitClient.send(cmd);
+  } catch (err) {
+    logger.debug({ err }, 'failed to get PR using prId');
+  }
+  return res;
+}
+
+export async function listRepositories(): Promise<ListRepositoriesOutput> {
+  const input: ListRepositoriesInput = {};
+  const cmd = new ListRepositoriesCommand(input);
+  return await codeCommitClient.send(cmd);
+}
+
+export async function createPrApprovalRule(
+  pullRequestId: string,
+  approvalRuleContent: string
+): Promise<CreatePullRequestApprovalRuleOutput> {
+  const input: CreatePullRequestApprovalRuleInput = {
+    approvalRuleContent,
+    approvalRuleName: 'Reviewers By Renovate',
+    pullRequestId,
+  };
+  const cmd = new CreatePullRequestApprovalRuleCommand(input);
+  return await codeCommitClient.send(cmd);
+}
+
+export function getCodeCommitUrl(
+  region: string,
+  repoName: string,
+  credentials: Credentials
+): string {
+  const signer = new aws4.RequestSigner(
+    {
+      service: 'codecommit',
+      host: `git-codecommit.${region}.amazonaws.com`,
+      method: 'GIT',
+      path: `v1/repos/${repoName}`,
+    },
+    credentials
+  );
+  const dateTime = signer.getDateTime();
+
+  /* istanbul ignore if */
+  if (!is.string(dateTime)) {
+    throw new Error(REPOSITORY_UNINITIATED);
+  }
+
+  const accessKeyId = credentials.accessKeyId;
+  const token: string = dateTime + 'Z' + signer.signature();
+
+  let username = `${accessKeyId}${
+    credentials.sessionToken ? `%${credentials.sessionToken}` : ''
+  }\n`;
+
+  // these modifications to username are only in case session token exists,
+  // and it's not supposed to work with renovate since it's a temporary token
+
+  // istanbul ignore if
+  if (username.includes('\n')) {
```

Comment:
why you add a newline above, when you remove it here?

Reply:
where is above?

the new line should be massaged because it ruins the URL

Reply:
mistake i think , cuz of too many changes, ill have to test it again without